### PR TITLE
KAFKA-16896 upgrade spotless version after we drop JDK8

### PR DESCRIPTION
**Pull request by m1a2st**
Jira: https://issues.apache.org/jira/browse/KAFKA-16896
update spotless version to 6.25.0 because we drop jdk8 in CI

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2832,7 +2824,7 @@ project(':streams:streams-scala') {
     dependsOn 'copyDependantLibs'
   }
 
-  // spotless 6.14 requires Java 11 at runtime
+  // spotless 6.25 requires Java 11 at runtime
   if (JavaVersion.current().isJava11Compatible()) {
```
Comment (mingyen066) [inline build.gradle]: Can we also OR Java 17 in this if condition?

File: build.gradle [hunk 2]
```diff
@@ -854,17 +851,12 @@ subprojects {
...
-        targetExclude('**/generated/**/*.java','**/generated-test/**/*.java')
-        importOrder('kafka', 'org.apache.kafka', 'com', 'net', 'org', 'java', 'javax', '', '\\#')
-        removeUnusedImports()
-      }
+  apply plugin: 'com.diffplug.spotless'
+  spotless {
+    java {
+      targetExclude('**/generated/**/*.java','**/generated-test/**/*.java')
+      importOrder('kafka', 'org.apache.kafka', 'com', 'net', 'org', 'java', 'javax', '', '\\#')
+      removeUnusedImports()
```
Comment (mingyen066) [inline build.gradle:859]: I think the limitation written in comment is still their. Can we use removeUnusedImports in JDK 21 already?

**Review by mingyen066 — COMMENTED** [review]
Thanks @m1a2st! I left some questions, PTAL.

File: build.gradle (hunk 1 above)
Comment (chia7712) [reply build.gradle]: this check is unnecessary since we assume the JDK 8 is dropped

File: build.gradle (hunk 2 above)
Comment (chia7712) [reply build.gradle:859]: > I think the limitation written in comment is still their. Can we use removeUnusedImports in JDK 21 already?

yes, I have tested it on my local. works well

**Comment by m1a2st** [comment]
@chia7712, @mingyen066 Thanks for your comments, I have been addressed by comments

**Review by chia7712 — APPROVED** [review]

**Comment by cadonna** [comment]
@m1a2st @chia7712 Have we already dropped JDK8? I still see `minJavaVersion = 8`, `sourceCompatibility = minJavaVersion`, and `targetCompatibility = minJavaVersionin` in `build.gradle`.  These variables should be changed to implement [KIP-750](https://cwiki.apache.org/confluence/x/P4vOCg). Should we not have had implemented KIP-750 before merging this PR?
This breaks builds with java 8.

**Comment by frankvicky** [comment]
Hi @cadonna,
I have opened a PR to address the issue you mentioned.
#17426

**Comment by chia7712** [comment]
@cadonna @cadonna, thanks for your response. We are planning to drop JDK 8, but we need this PR first since we need CI builds under JDK 21. However, spotless can’t run with JDK 21, which means our CI has been running without a spotless check for some time, making build errors more likely.

I’ve created tasks for the JDK 8 removal. @frankvicky is working on build.gradle and code cleanup, @Yunyung will update the E2E JDK and clean up old and unsupported Kafka versions, and @m1a2st is going to update the documentation for JDK 8. You can find more details in KAFKA-12894.

In short, we’re getting ready to say goodbye to JDK 8 😃.
